### PR TITLE
Introduce `stable` tag on containers

### LINKED ISSUE #488: (Feature request) Introduce stable tag on docker hub
### Describe feature

According to your documentation it is not recommended to use the latest tag for docker installations, as that can contain unstable minor versions.

I am using watchtower as a way to automatically update all docker containers, this will not work when using specific version tags.

Can a stable tag be introduced, which will always contain the latest stable version?

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by perfectra1n**
Closes #488

### CODE REVIEW

File: .github/workflows/main-docker.yml [hunk 1]
```diff
@@ -242,6 +242,17 @@ jobs:
...
+          # If the ref is a tag, also tag the image as stable as this is part of a 'release'
+          if [[ "${GITHUB_REF}" == refs/tags/* ]]; then
+            docker buildx imagetools create $(jq -cr '.tags | map("-t " + .) | join(" ")' <<< "$DOCKER_METADATA_OUTPUT_JSON") \
+              -t ${{ env.GHCR_REGISTRY }}/${{ env.IMAGE_NAME }}:stable \
+              $(printf '${{ env.GHCR_REGISTRY }}/${{ env.IMAGE_NAME }}@sha256:%s ' *)
+            
+            docker buildx imagetools create $(jq -cr '.tags | map("-t " + .) | join(" ")' <<< "$DOCKER_METADATA_OUTPUT_JSON") \
+              -t ${{ env.DOCKERHUB_REGISTRY }}/${{ env.IMAGE_NAME }}:stable \
+              $(printf '${{ env.DOCKERHUB_REGISTRY }}/${{ env.IMAGE_NAME }}@sha256:%s ' *)
+          fi
```
Comment (eliandoran) [inline .github/workflows/main-docker.yml:255]: You did quite a bit of bash magic in here. 😎 

Does this also handle the case where I push a beta version such as `v0.90.7-beta`? In this case we shouldn't tag it as `stable`; the `stable` tag should remain to `v0.90.4` until say `v0.90.8` comes along.

Comment (perfectra1n) [reply .github/workflows/main-docker.yml:255]: Yeah it's not my favorite - and no, that's a good point. I was just catching if there was a new tag pushed. I'm assuming I would just have to catch if there is a `-` in the string of the tag that is pushed, to assume that it's not a `stable` build?

File: .github/workflows/main-docker.yml [hunk 2]
```diff
@@ -242,6 +242,18 @@ jobs:
           docker buildx imagetools create $(jq -cr '.tags | map("-t " + .) | join(" ")' <<< "$DOCKER_METADATA_OUTPUT_JSON") \
             -t ${{ env.DOCKERHUB_REGISTRY }}/${{ env.IMAGE_NAME }}:${REF_NAME} \
             $(printf '${{ env.DOCKERHUB_REGISTRY }}/${{ env.IMAGE_NAME }}@sha256:%s ' *)
+
+          # If the ref is a tag, also tag the image as stable as this is part of a 'release'
+          # and only go in the `if` if there is NOT a `-` in the tag's name, due to tagging of `-alpha`, `-beta`, etc...
+          if [[ "${GITHUB_REF}" == refs/tags/* && ! "${REF_NAME}" =~ - ]]; then
```
Comment (perfectra1n) [inline .github/workflows/main-docker.yml:248]: This `if` should now appropriately be skipped if a `-` is in the tag's name.

**Review by eliandoran — APPROVED** [review]
LGTM.

**Pull request merged**